File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    stack = []
    answer = 0
    for i in ingredient:
        stack.append(i)
        if len(stack) >= 4:
            if stack[-4] == 1 and stack[-3] == 2 and stack[-2] == 3 and stack[-1] == 1:
                for _ in range(4):
                    stack.pop()
                answer += 1
    return answer

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.swift
import Foundation

func solution(_ ingredient:[Int]) -> Int {
    var stack = [Int]()
    var result = 0
    
    for i in ingredient {
        stack.append(i)
        let count = stack.count
        if count >= 4
        && stack[(count - 4)...(count - 1)] == [1, 2, 3, 1] {
            result += 1
            stack.popLast()
            stack.popLast()
            stack.popLast()
            stack.popLast()
        }
    }
    
    return result
}